tensorflow_probability/python/distributions/empirical_test.py

```diff
@@ -0,0 +1,694 @@
+
+  def testShapes(self):
+    for samples_shape in ([2], [2, 4], [4, 2, 4]):
+      input_ = random_samples(samples_shape)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=samples_shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph)
+      self.assertAllEqual(samples_shape[:-1],
+                          self.evaluate(dist.batch_shape_tensor()))
+      self.assertAllEqual([],
+                          self.evaluate(dist.event_shape_tensor()))
+
+  def testCdfNoBatch(self):
+    sample = [0, 1, 1, 2]
+    events = [
+        [1],
+        [0, 1],
+        [[0, 1], [1, 2]]
+    ]
+    expected_cdfs = [
+        [0.75],
+        [0.25, 0.75],
+        [[0.25, 0.75], [0.75, 1]]
+    ]
+
+    for event, expected_cdf in zip(events, expected_cdfs):
+      input_ = tf.convert_to_tensor(sample, dtype=np.float32)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=input_.shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph)
+      self.assertAllClose(self.evaluate(dist.cdf(event)),
+                          expected_cdf)
+      self.assertAllClose(self.evaluate(dist.log_cdf(event)),
+                          np.log(expected_cdf))
+
+  def testCdfWithBatch(self):
+    sample = [[0, 0, 1, 2], [0, 1, 1, 2]]
+    events = [
+        [0],
+        [0, 1],
+        [[0, 1], [1, 2]]
+    ]
+    expected_cdfs = [
+        [0.5, 0.25],
+        [0.5, 0.75],
+        [[0.5, 0.75], [0.75, 1]]
+    ]
+
+    for event, expected_cdf in zip(events, expected_cdfs):
+      input_ = tf.convert_to_tensor(sample, dtype=np.float32)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=input_.shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph)
+      self.assertAllClose(self.evaluate(dist.cdf(event)),
+                          expected_cdf)
+      self.assertAllClose(self.evaluate(dist.log_cdf(event)),
+                          np.log(expected_cdf))
+
+  def testPmfNoBatch(self):
+    sample = [0, 1, 1, 2]
+    events = [
+        [0],
+        [0, 1],
+        [[0, 1], [1, 2]]
+    ]
+    expected_pmfs = [
+        [0.25],
+        [0.25, 0.5],
+        [[0.25, 0.5], [0.5, 0.25]]
+    ]
+
+    for event, expected_pmf in zip(events, expected_pmfs):
+      input_ = tf.convert_to_tensor(sample, dtype=np.float32)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=input_.shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph)
+      self.assertAllClose(self.evaluate(dist.prob(event)),
+                          expected_pmf)
+
+  def testPmfWithBatch(self):
+    sample = [[0, 0, 1, 2], [0, 1, 1, 2]]
+    events = [
+        [0],
+        [0, 1],
+        [[0, 1], [1, 2]]
+    ]
+    expected_pmfs = [
+        [0.5, 0.25],
+        [0.5, 0.5],
+        [[0.5, 0.5], [0.25, 0.25]]
```

Comment:
similar comment to above, avoid some ambiguity by modifying samples to different values per batch entry, maybe just make the second batch entry 10x scale.